qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -216,6 +199,119 @@ def _unpack_num_multi_sigs(self, num, sigs):
+                # for the associated Cliffords
+                if isclose(inst.params[0], np.pi) or isclose(inst.params[0], -np.pi):
+                    map_index = "z"
+                elif isclose(inst.params[0], np.pi / 2):
+                    map_index = "s"
+                elif isclose(inst.params[0], -np.pi / 2):
+                    map_index = "sdg"
+                else:
+                    raise QiskitError("wrong param {} for rz in clifford".format(inst.params[0]))
+            else:
+                raise QiskitError(
+                    "Instruction {} could not be converted to Clifford gate".format(name)
+                )
+
+        return single_gate_map[(map_index, str(qubits))]
+
+    @classmethod
+    def compose_num_with_clifford(
+        cls, num_qubits: int, composed_num: int, qc: QuantumCircuit, basis_gates: List[str]
+    ) -> int:
+        """Compose a number that represents a Clifford, with a single-gate Clifford, and return the
+        number that represents the resulting Clifford."""
+
+        # The numbers corresponding to single gate Cliffords are not in sequence -
+        # see num_from_1q_clifford_single_gate. To compute the index in
+        # the array CLIFF_COMPOSE_DATA_1Q, we map the numbers to [0, 8].
+        n = num_qubits
+        single_gate_map = CLIFF_SINGLE_GATE_MAP_1Q if n == 1 else CLIFF_SINGLE_GATE_MAP_2Q
+        cliff_compose_data = CLIFF_COMPOSE_DATA_1Q if n == 1 else CLIFF_COMPOSE_DATA_2Q
+        map_clifford_num_to_array_index = {}
+        num_single_gate_cliffs = len(single_gate_map)
+        for k in list(single_gate_map):
+            map_clifford_num_to_array_index[single_gate_map[k]] = list(
+                single_gate_map.keys()
+            ).index(k)
+        if n == 1:
+            for inst, qargs, _ in qc:
+                num = cls.num_from_clifford_single_gate(
+                    inst=inst, qubits=[0], rb_num_qubits=1, basis_gates=basis_gates
+                )
+                index = num_single_gate_cliffs * composed_num + map_clifford_num_to_array_index[num]
+                composed_num = cliff_compose_data[index]
+        else:
+            for inst, qargs, _ in qc:
+                if inst.num_qubits == 2:
+                    qubits = [qc.find_bit(qargs[0]).index, qc.find_bit(qargs[1]).index]
+                else:
+                    qubits = [qc.find_bit(qargs[0]).index]
+                num = cls.num_from_clifford_single_gate(
+                    inst=inst, qubits=qubits, rb_num_qubits=2, basis_gates=basis_gates
+                )
+                index = num_single_gate_cliffs * composed_num + map_clifford_num_to_array_index[num]
+                composed_num = cliff_compose_data[index]
+        return composed_num
+
+    @classmethod
+    def clifford_inverse_by_num(cls, num: int, num_qubits: int):
+        """Return the number of the inverse Clifford to the input num"""
+        if num_qubits == 1:
+            return CLIFF_INVERSE_DATA_1Q[num]
```

Comment:
As before, it might be better to define at the beginning
```
CLIFF_INVERSE_DATA = {1:CLIFF_INVERSE_DATA_1Q, 2:CLIFF_INVERSE_DATA_2Q}
```

and then do here
```
return CLIFF_INVERSE_DATA[num_qubits][num]
```

Reply:
I actually took this template and applied it for all three:
``` 
CLIFF_SINGLE_GATE_MAP = {1: CLIFF_SINGLE_GATE_MAP_1Q, 2: CLIFF_SINGLE_GATE_MAP_2Q}
CLIFF_COMPOSE_DATA = {1: CLIFF_COMPOSE_DATA_1Q, 2: CLIFF_COMPOSE_DATA_2Q}
CLIFF_INVERSE_DATA = {1: CLIFF_INVERSE_DATA_1Q, 2: CLIFF_INVERSE_DATA_2Q}
```